src/lib/wallet.js

```diff
@@ -55,13 +59,55 @@ async function getKeyPair() {
 }
 
 async function initContractInstances() {
-  if (!store.getters.mainnet && !process.env.RUNNING_IN_TESTS) return;
+  if (!store.getters.mainnet && !process.env.RUNNING_IN_TESTS && process.env.NETWORK !== 'Testnet')
+    return;
   const contractAddress = await store.dispatch('getTipContractAddress');
-  store.commit(
-    'setTipping',
-    await store.state.sdk.getContractInstance(TIPPING_CONTRACT, {
-      contractAddress,
-      forceCodeCheck: true,
+  const contractAddressV2 = await store.dispatch('getTipContractAddressV2');
+  const contractInstance = await store.state.sdk.getContractInstance(TIPPING_CONTRACT, {
+    contractAddress,
+    forceCodeCheck: true,
+  });
+  const contractInstanceV2 = await store.state.sdk.getContractInstance(TIPPING_CONTRACT_V2, {
+    contractAddress: contractAddressV2,
+    forceCodeCheck: true,
+  });
+  store.commit('setTipping', contractInstance);
+  store.commit('setTippingV2', contractInstanceV2);
+}
+
+async function tokenBalance(token, address) {
+  const tokenContract = await store.state.sdk.getContractInstance(FUNGIBLE_TOKEN_CONTRACT, {
+    contractAddress: token,
+  });
+
+  const { decodedResult } = await tokenContract.methods.balance(address);
+  return new BigNumber(decodedResult || 0).toFixed();
+}
+
+async function loadTokenBalances(address) {
+  const tokens = await Backend.getTokenBalances(address);
+  await Promise.all(
+    Object.entries(tokens).map(async token => {
+      const balance = await tokenBalance(token[0], address);
+      const convertedBalance = parseFloat(convertToken(balance, -token[1].decimals)).toFixed(2);
+      const infoIndex = store.state.tokenInfo.findIndex(({ contract }) => contract === token[0]);
```

Comment:
"heavy" find, structure from backend is already optimized for more simple access

Reply:
The conversion is done so that the structure is suitable for usage with dropdown control. It accepts an array with "text" and "value" fields